Update scheduling queries and unit tests to conform to Aerie v2.6

joshhaug opened the pull request:
The primary motivation for this PR is to update the scheduling module to use the new db schemas introduced in Aerie v2.6.0. More information on these changes is available [on the Aerie upgrade docs](https://nasa-ammos.github.io/aerie-docs/upgrade-guides/2-5-0-to-2-6-0/).

Additionally, there are also some cleanup tasks that I've lumped in:
* Replaced the default banananation 2.2 jar with the latest banananation 2.6 jar, and updates the `.env` file accordingly
* Refactored to replace the deprecated `datetime.utcnow()` calls with the new timezone-aware `datetime.now(timezone.utc)`. This is not a necessary change, but prevents us from throwing deprecation errors.

I'm happy to split these up into separate PRs if preferred.

joshhaug commented:
Looks like this is failing integration tests because we no longer have the 2.2.0 banananation jar. I can add this file back in, or we can update the tests to use the newest jar. I'm leaning towards the latter.

cartermak commented on src/aerie_cli/aerie_client.py:
```diff
@@ -1488,30 +1488,25 @@ def get_scheduling_goals_by_specification(self, spec_id):
         obj = dict()
-        obj["name"] = name
-        obj["model_id"] = model_id
+        #obj["metadata"] = {"data": {"name": name, "models_using": {"data": {"model_id": model_id}}}}
+        #obj["name"] = name
+        #obj["model_id"] = model_id
```
Remove

cartermak commented on src/aerie_cli/aerie_client.py:
```diff
@@ -1488,30 +1488,25 @@ def get_scheduling_goals_by_specification(self, spec_id):
-                    description
-                    author
-                    last_modified_by
-                    created_date
-                    modified_date
                 }
```
Does it break anything to exclude all of this information that used to be included?

cartermak commented on src/aerie_cli/aerie_client.py:
```diff
@@ -1527,16 +1522,24 @@ def upload_scheduling_goals(self, upload_object):
         ]
         """
 
+        # Note that as of Aerie v2.3.0, the metadata (incl. model_id and goal name) are stored in a separate table
+        formatted_upload_objects = []
+        for entry in upload_object:
+            new_entry = {}
+            new_entry["definition"] = entry["definition"]
+            new_entry["metadata"] = {"data": {"name": entry["name"], "models_using": {"data": {"model_id": entry["model_id"]}}}}
+            formatted_upload_objects.append(new_entry)
+        
```
For long-term maintainability, would it be better to break the format of `upload_object` than to hard-code this mapping? Who are the users of this method?

joshhaug replied:
The only user of this method is `scheduling.upload`. Is it preferable to deliver the `upload_object` to this method as it should be uploaded?

cartermak replied:
I think that's preferable. I'd say we should avoid breaking changes to the API, but I don't know of anyone using the scheduling rule Python API.

cartermak reviewed (commented):
1. I suggest you update the integration test defaults to also use 2.6.0 [here](https://github.com/NASA-AMMOS/aerie-cli/blob/353ffbfa04973d85afc11cb4a7e733d0136d0109/tests/integration_tests/conftest.py#L44). (I forgot to put that in the integration test documentation!)
2. I think what's breaking the integration tests now is that, since Aerie 2.4.0ish, Aerie-CLI is no longer able to tell if the Aerie host has auth enabled. [This](https://github.com/NASA-AMMOS/aerie-cli/blob/353ffbfa04973d85afc11cb4a7e733d0136d0109/src/aerie_cli/aerie_host.py#L239-L257) is the method that used to work and needs to be updated.

joshhaug commented:
Update: I've modified this PR to also include updates to the constraint queries to bring them up to parity with the changes in [v2.6.0](https://nasa-ammos.github.io/aerie-docs/upgrade-guides/2-5-0-to-2-6-0/). Also updated to fix the `is_auth_enabled()` function -- for some reason Aerie now presents the "Authentication is disabled" message on posting to the `/auth/login` endpoint instead of `/auth/session`. Updated this PR accordingly.

cartermak commented on src/aerie_cli/aerie_client.py:
```diff
@@ -1488,49 +1488,53 @@ def get_scheduling_goals_by_specification(self, spec_id):
     def upload_scheduling_goal(self, model_id, name, definition):
         obj = dict()
-        obj["name"] = name
-        obj["model_id"] = model_id
         obj["definition"] = definition
-
         return self.upload_scheduling_goals([obj])
```
Does this need to build the `update_scheduling_goals` object structure?

joshhaug replied:
Tbh I think we can remove this method entirely, it is not being used anywhere.

cartermak replied:
Even better!

joshhaug replied:
Updated in 1d42f2df28796f48635a9953580ac9829dd104f0 to remove `upload_scheduling_goal`, as `upload_scheduling_goals` can just be used directly if needed.

cartermak reviewed (approved):
LGTM. Feel free to merge once CI passes
